src/3p.js

```diff
@@ -109,18 +109,25 @@ function executeAfterWriteScript(win, fn) {
  * @param {string} src
  */
 export function validateSrcPrefix(prefix, src) {
-
+  const validPrefixes = [];
   if (!isArray(prefix)) {
     prefix = [prefix];
   }
-
-  for (const p of prefix) {
-    if (src.indexOf(p) === 0) {
+  const totalPrefixes = prefix.length;
+  if (src !== undefined) {
+    let p = 0;
+    for (p = 0; p <= totalPrefixes; p++) {
+      const protocolIndex = src.toLowerCase().indexOf(prefix[p]);
```

Comment:
This action is only lower-casing to validate against the **protocol**/prefix not the entire path, A User agent should not treat HTTPS:// and https:// differently. Ideally both the prefix[p] and the source should be lower-cased then validated for protocol.

Removing on next commit.